tests/Unit/API/Google/MerchantReportTest.php

```diff
@@ -0,0 +1,182 @@
+		$product_view->setId( $mc_id );
+		return $product_view;
+	}
+
+	public function test_get_product_view_report() {
+		$test_merchant_id = 432;
+		$wc_product_id_1  = 882;
+		$wc_product_id_2  = 883;
+		$page_size        = 800;
+
+		add_filter(
+			'woocommerce_gla_product_view_report_page_size',
+			function () use ( $page_size ) {
+				return $page_size;
+			}
+		);
+
+		$this->options->method( 'get_merchant_id' )->willReturn( $test_merchant_id );
+
+		$this->product_helper->method( 'get_wc_product_id' )->will(
+			$this->returnCallback(
+				function ( $mc_id ) use ( $wc_product_id_1, $wc_product_id_2 ) {
+					if ( $mc_id === 'online:en:ES:gla_' . $wc_product_id_1 ) {
+							return $wc_product_id_1;
+					}
+
+					if ( $mc_id === 'online:en:ES:gla_' . $wc_product_id_2 ) {
+						return $wc_product_id_2;
+					}
+
+					return 0;
+				}
+			)
+		);
+
+		$product_view_1 = $this->create_product_view_product_status( 'online:en:ES:gla_' . $wc_product_id_1 );
+		$product_view_2 = $this->create_product_view_product_status( 'online:en:ES:gla_' . $wc_product_id_2, 'NOT_ELIGIBLE_OR_DISAPPROVED' );
+		$product_view_3 = $this->create_product_view_product_status( 'online:en:ES:external' );
+
+		$report_row_1 = new ReportRow();
+		$report_row_1->setProductView( $product_view_1 );
+
+		$report_row_2 = new ReportRow();
+		$report_row_2->setProductView( $product_view_2 );
+
+		$report_row_3 = new ReportRow();
+		$report_row_3->setProductView( $product_view_3 );
+
+		$response = $this->createMock( SearchResponse::class );
+		$response->expects( $this->once() )
+			->method( 'getResults' )
+			->willReturn( [ $report_row_1, $report_row_2, $report_row_3 ] );
+
+		$response->expects( $this->once() )
+			->method( 'getNextPageToken' )
+			->willReturn( null );
+
+		$search_request = new SearchRequest();
+		$search_request->setQuery(
+			'SELECT product_view.id,product_view.expiration_date,product_view.aggregated_destination_status FROM ProductView'
+		);
+
+		$search_request->setPageSize( $page_size );
+
+		$this->shopping_client->reports->expects( $this->once() )
+			->method( 'search' )
+			->with( $test_merchant_id, $search_request )
+			->willReturn( $response );
+
+		$this->assertEquals(
+			[
+				'statuses'        => [
+					$wc_product_id_1 => [
+						'product_id'      => $wc_product_id_1,
+						'status'          => MCStatus::APPROVED,
+						'expiration_date' => $this->convert_shopping_content_date( $product_view_1->getExpirationDate() ),
+					],
+					$wc_product_id_2 => [
+						'product_id'      => $wc_product_id_2,
+						'status'          => MCStatus::DISAPPROVED,
+						'expiration_date' => $this->convert_shopping_content_date( $product_view_2->getExpirationDate() ),
+					],
+				],
+				'next_page_token' => null,
+			],
+			$this->merchant_report->get_product_view_report()
+		);
+	}
+
+	public function test_get_product_view_report_with_exception() {
+		$this->options->method( 'get_merchant_id' )->willReturn( 432 );
```

Comment:
We're also setting the return value for `get_merchant_id` on the options mock in `test_get_product_view_report` but beyond that we don't do anything with it so instead of repeating this had you considered creating a class property for it and then assigning it to the mock in `setUp`?

Reply:
Thanks for the suggestion, I added here: https://github.com/woocommerce/google-listings-and-ads/pull/2261/commits/6a9a0f96d577930181961b6c7fed1587a6225513